### ISSUE TITLE
Semantic conveyed through Indented UI and Reflow 1.4.10

### ISSUE THREAD

**Issue by cscrosati** [w3c/wcag#887]
Could it be that indented UI where the indent has meaning qualifies as an exception in the 1.4.10 Reflow requirement? Indentation conveys a lot of meaning visually that isn't color based, so it seems like this is a suitable exception when used as part of a control or UI that uses indentation.

I believe this is covered by the sentence 

> "Except for parts of the content which require two-dimensional layout for usage or meaning."

but I want to confirm this is correct in my interpretation. It might also be good to add this to the exception examples.

Here are a couple examples I have encountered recently:

**Example 1**: Indented List (or could be a system file picker, or a TreeView control)
- This is level 1
  - This is level 2
    - This is level 3
      - This is level 4
        - This is level 5
          - This is level 6
            - This is level 7
              - Eventually deeper levels get quite indented. Making them flat level removes a lot of context that is visually conveyed without color change needed, nor text styling specifics.

**Example 2**: Sample code as text, imagine this gets very deeply indented, like in Chrome Dev tools Elements tab for example.
````
<html>
    <head>
    </head>
    <body>
        <div>
            <!-- ...... keep indenting more stuff in here -->
        </div>
    </body>
</html>
````

Note that for example 2 this could be managed by allowing the overflowing indented content to return to the beginning of the content area, but then the indent itself might be overflowing, or the content text becomes so long that it becomes difficult to identify the level of indentation without referring to the text backward from the current point of reading. You can understand better what I'm typing here by opening the Chrome Dev tools Elements pane in a moderately complex web page, with the Word Wrap setting enabled. Things get lost at deeper level.

Thoughts?

**Comment by JAWS-test** [comment]
I think the examples fall under mentioned the exception: "require two-dimensional layout for usage or meaning"

**Comment by mraccess77** [comment]
This discussion came up with code samples as well and I believe there are ways to make a pre tag wrap as well as the list items should be able to wrap.  As for indention -- a settings for different indention spacing or identifiers could be used -- so I'm not sure (personal opinion) if it falls into the exception.

**Comment by WayneEDick** [comment]
To All,
Lists are a problem. The problem is that indentation in the large print
realm is less valuable than space.

There are a couple of ways to handle this that meet 1.4.10.


   1. Use a smaller indentation as the CSS pixel width decreases. That is
   at each breakpoint decide: what is a visible and non-disruptive indentation
   size. Remember 1 em is a hell of a lot more visible at 400% than it at 100%.
   2. Require users to move list items to the left margin to see entire
   lines, but never require users to horizontally scroll text within a list
   item.

That should do it. The key thing to avoid is line-by-line horizontal
scrolling. Those are the scrolls that cost enormous operational overhead.

Best, Wayne

**Comment by JAWS-test** [comment]
There are contents that have to be scrolled already without zoom. These can be the above examples, but also tables and other exceptions mentioned in 1.4.10. At 400% they have to be scrolled 4x more (if no layout changes take place). The question now would be: Is it a violation of 1.4.10 that content to be scrolled has to be scrolled 4x more? I think, according to the wording of 1.4.10: no. Although according to the intentions of the WCAG it should be: Yes.

**Comment by patrickhlauke** [comment]
> There are contents that have to be scrolled already without zoom

but those are not covered by the SC. more specifically: the SC only cares about (currently) when the viewport/browser window is 320 CSS px for a page that is primarily vertical-scrolling content / 256 CSS px for a page that is primarily horizontal-scrolling content.

if stuff already causes two-directional scrolling at larger other viewport sizes, that's not the current concern of the SC (however, as it's likely that if it already requires bi-directional scrolling in a large viewport, it will also require scrolling - even more or not - at zoom / smaller viewport, it's still a failure anyway - just that the failure is framed as "at the small viewport, this happens")

In theory, a page could have bi-directional scrolling happening all the time, EXCEPT at the very specific 320 CSS px width or 256 CSS px height. like a super-specific media query. that is, of course, a loophole. see https://github.com/w3c/wcag/issues/698

**Comment by jake-abma** [comment]
besides the loophole, if the indent has meaning where reflow is not possible / looses the meaning, it's an exception indeed.

**Comment by alastc** [comment]
If the indentation is "required ... for usage or meaning", then it would fit the exception. However, if there are valid techniques for showing the meaning without (as much) indentation then it wouldn't be required.

Wayne has discussed and sent in such a technique (which I have in my backlog to put into github and raise on a survey).

We know there are techniques for dealing with things like tables and code markup in better ways that aren't currently required by reflow, so it's an area we can tighten up on in future.

**Comment by mraccess77** [comment]
So are code blocks exempted from 1.4.10 or not?

**Comment by patrickhlauke** [comment]
> So are code blocks exempted from 1.4.10 or not?

you can force a particular line break/presentation of code blocks without altering their actual structure/indentation when copy-pasting, but for languages where indentation actually carries meaning (like python) it might still make it unreadable/confusing for somebody just looking at the code block directly on the page (rather than copy/pasting it into anything)

**Comment by patrickhlauke** [comment]
put succinctly, yes (in my view)

**Comment by alastc** [comment]
The understanding doc has a section for [content exceptions](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html#content-exceptions-for-reflow). Perhaps we could add something like:
<hr>
The presentation of code where the layout has specific meaning (such as indentation for Python) would loose meaning if that layout were not presented correctly. Therefore this Success Criterion does not apply where that meaning would be lost. However, this is not the case for many languages where wrapping can be applied without loosing meaning.

**Comment by scottaohara** [comment]
this issue helped guide some of the updated content for the [Reflow understanding doc](https://deploy-preview-4055--wcag2.netlify.app/understanding/reflow) and the creation of the new G224 technique.  Closing this as resolved. Please reopen or file a new issue if questions remain.
